scipy/interpolate/tests/test_interpolate.py

```diff
@@ -2613,6 +2613,28 @@ def test_broadcastable_input(self):
                          interp_points3.size ==
                          interp_points4.size, True)
 
+    def test_read_only(self):
+        # input data
+        np.random.seed(0)
+        xy = np.random.random((10, 2))
+        x, y = xy[:, 0], xy[:, 1]
+        z = np.hypot(x, y)
+        
+        # x-y grid for interpolation
+        X = np.linspace(min(x), max(x))
+        Y = np.linspace(min(y), max(y))
+        X, Y = np.meshgrid(X, Y)
+
+        xy.setflags(write=False)
+        z.setflags(write=False)
+        X.setflags(write=False)
+        Y.setflags(write=False)
+
+        for interpolator in (NearestNDInterpolator, LinearNDInterpolator,
+                             CloughTocher2DInterpolator):
+            interp = interpolator(xy, z)
+            interp(X, Y)
```

Comment:
This still isn't testing `xi` properly because you need to pass a single array here as well. Otherwise, a new `xi` array is allocated internally here:
https://github.com/scipy/scipy/blob/0765482e9a7cd7a7975f6161d2ab91994762ac8e/scipy/interpolate/interpnd.pyx#L173-L175

Reply:
I missed it. Thanks. PTAL again.